tests/Doctrine/Tests/DBAL/Platforms/AbstractMySQLPlatformTestCase.php

```diff
@@ -946,4 +948,75 @@ public function testGetCreateTableSQLWithColumnCollation() : void
             'Column "no_collation" will use the default collation from the table/database and "column_collation" overwrites the collation on this column'
         );
     }
+
+    /**
+     * @group DBAL-2576
+     */
+    protected function createNotDuplicateDropForeignKeySql()
+    {
+        $diff = new TableDiff('documents');
+
+        // Create offices table
+        $offices = new Table('offices');
+        $offices->addColumn('id', 'integer');
+
+        // Create documents table
+        $documents = new Table('documents');
+        $documents->addColumn('id', 'integer');
+        $documents->addColumn('office_id', 'integer');
+        $documents->addForeignKeyConstraint('offices', [ 'office_id' ], [ 'id' ], [ 'onDelete' => 'CASCADE' ], 'FK_a788692ffa0c224');
+        $documents->setPrimaryKey([ 'id' ]);
+
+        // Added foreign key
+        $diff->addedForeignKeys['FK_a788692ffa0c224'] = $documents->getForeignKey('FK_a788692ffa0c224');
```

Comment:
This is not what real DBAL usage looks like. The comparator should be used to produce the diff of two given schemas, the diff shouldn't be built in parallel with the schema(s).